src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/GrTagsForHydro.hpp

```diff
@@ -0,0 +1,147 @@
+ *   * `Tags::CoordinateMap<Dim, Frame::Grid, Frame::Inertial>`
+ *   * `domain::Tags::FunctionsOfTime`
+ *   * `evolution::dg::subcell::Tags::Coordinates<Dim, Frame::Logical>`
+ *   * `::Tags::AnalyticSolutionOrData`
+ *
+ * DataBox changes:
+ * - Adds:
+ *   * `dg::subcell::Tags::Inactive<System::spacetime_variables_tag>`
+ *   * `Tags::OnSubcellFaces<System::flux_spacetime_variables_tag, Dim>`
+ *
+ * - Removes: nothing
+ * - Modifies: nothing
+ *
+ * \note This action relies on the `SetupDataBox` aggregated initialization
+ * mechanism, so `Actions::SetupDataBox` must be present in the
+ * `Initialization` phase action list prior to this action.
+ */
+template <typename System, size_t Dim>
+struct GrTagsForHydro {
+  using initialization_tags = tmpl::list<Initialization::Tags::InitialTime>;
+
+  using gr_tag = typename System::spacetime_variables_tag;
+  using subcell_gr_tag = evolution::dg::subcell::Tags::Inactive<gr_tag>;
+  using subcell_faces_gr_tag = evolution::dg::subcell::Tags::OnSubcellFaces<
+      typename System::flux_spacetime_variables_tag, Dim>;
+  using GrVars = typename gr_tag::type;
+  using SubcellGrVars = typename subcell_gr_tag::type;
+  using FaceGrVars = typename subcell_faces_gr_tag::type::value_type;
+
+  using return_tags = tmpl::list<subcell_gr_tag, subcell_faces_gr_tag>;
+  using argument_tags =
+      tmpl::list<Initialization::Tags::InitialTime,
+                 evolution::dg::subcell::Tags::Mesh<Dim>,
+                 domain::Tags::ElementMap<Dim, Frame::Grid>,
+                 domain::CoordinateMaps::Tags::CoordinateMap<Dim, Frame::Grid,
+                                                             Frame::Inertial>,
+                 domain::Tags::FunctionsOfTime,
+                 evolution::dg::subcell::Tags::Coordinates<Dim, Frame::Logical>,
+                 ::Tags::AnalyticSolutionOrData>;
+
+  template <typename AnalyticDataOrSolution>
+  static void apply(
+      const gsl::not_null<SubcellGrVars*> cell_centered_gr_vars,
+      const gsl::not_null<std::array<FaceGrVars, Dim>*> face_centered_gr_vars,
+      const double initial_time, const Mesh<Dim>& subcell_mesh,
+      const ElementMap<Dim, Frame::Grid>& logical_to_grid_map,
+      const domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, Dim>&
+          grid_to_inertial_map,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+          functions_of_time,
+      const tnsr::I<DataVector, Dim, Frame::Logical>&
+          subcell_logical_coordinates,
+      const AnalyticDataOrSolution& analytic_data_or_solution) noexcept {
+    const size_t num_grid_points = subcell_mesh.number_of_grid_points();
+    const auto cell_centered_inertial_coords =
+        grid_to_inertial_map(logical_to_grid_map(subcell_logical_coordinates),
+                             initial_time, functions_of_time);
+
+    // Set cell-centered vars. Need to first do without prefix then move into
```

Comment:
why?

Reply:
because calling `assign_subset` with the prefixes didn't work

Reply:
that's silly but not something you need to address...